packages/world/ts/config/v2/world.test.ts

```diff
@@ -0,0 +1,879 @@
+          Second: {
+            schema: { secondKey: "Static", secondName: "Dynamic", secondAge: "uint256" },
+            primaryKey: ["secondKey", "secondAge"],
+          },
+        },
+        userTypes: { Static: "address", Dynamic: "string" },
+      });
+      const expected = {
+        tables: {
+          First: {
+            schema: {
+              firstKey: {
+                type: "address",
+                internalType: "Static",
+              },
+              firstName: {
+                type: "string",
+                internalType: "Dynamic",
+              },
+              firstAge: {
+                type: "uint256",
+                internalType: "uint256",
+              },
+            },
+            keySchema: {
+              firstKey: {
+                type: "address",
+                internalType: "Static",
+              },
+              firstAge: {
+                type: "uint256",
+                internalType: "uint256",
+              },
+            },
+            valueSchema: {
+              firstName: {
+                type: "string",
+                internalType: "Dynamic",
+              },
+            },
+            primaryKey: ["firstKey", "firstAge"],
+          },
+          Second: {
+            schema: {
+              secondKey: {
+                type: "address",
+                internalType: "Static",
+              },
+              secondName: {
+                type: "string",
+                internalType: "Dynamic",
+              },
+              secondAge: {
+                type: "uint256",
+                internalType: "uint256",
+              },
+            },
+            keySchema: {
+              secondKey: {
+                type: "address",
+                internalType: "Static",
+              },
+              secondAge: {
+                type: "uint256",
+                internalType: "uint256",
+              },
+            },
+            valueSchema: {
+              secondName: {
+                type: "string",
+                internalType: "Dynamic",
+              },
+            },
+            primaryKey: ["secondKey", "secondAge"],
+          },
+        },
+        userTypes: { Static: "address", Dynamic: "string" },
+        enums: {},
+        namespace: "",
+        namespaces: {},
+      } as const;
+      attest<typeof expected>(config);
+    });
+
+    it("should throw if referring to fields of different tables", () => {
+      attest(
+        resolveWorldConfig({
+          tables: {
+            First: {
+              schema: { firstKey: "address", firstName: "string", firstAge: "uint256" },
+              primaryKey: ["firstKey", "firstAge"],
+            },
+            Second: {
+              schema: { secondKey: "address", secondName: "string", secondAge: "uint256" },
+              // @ts-expect-error Type '"firstKey"' is not assignable to type '"secondKey" | "secondAge"'
+              primaryKey: ["firstKey", "secondAge"],
```

Comment:
do we have/want/need any checks around "unique fields" defined here so we don't allow for `["firstKey", "firstKey"]`?

Reply:
Don't have it yet, unclear if it's possible at a type level, but will ask david about ideas. Maybe we can map the tuple to an object that has the tuple values as keys, then convert it back to a tuple and check the length? Would also keep this one for later though.